site/app/libraries/database/DatabaseQueries.php

```diff
@@ -9372,6 +9399,44 @@ public function checkNonMappedUsers() {
         return $this->rowsToArray($this->submitty_db->rows());
     }
 
+    /**
+     * Adds a component grader to the current_gradable_grader table
+     *
+     * @param \app\models\gradeable\Component $component
+     * @param bool $isTeam
+     * @param string $grader_id
+     * @param string $graded_id
+     * @return void
+     */
+    public function addComponentGrader($component, $isTeam, $grader_id, $graded_id) {
+            $this->course_db->query("
+            INSERT INTO active_graders (gc_id, grader_id, ag_user_id, ag_team_id, timestamp)
+            VALUES (?, ?, ?, ?, ?) ON CONFLICT DO NOTHING
+        ", [$component->getId(), $grader_id, $isTeam ? null : $graded_id, $isTeam ? $graded_id : null, $this->core->getDateTimeNow()]);
+    }
+
+    /**
+     * Removes a component grader to the active graders table
+     * @param \app\models\gradeable\Component $component
+     * @param bool $isTeam
+     * @param string $grader_id
+     * @param string $graded_id
+     * @return void
+     */
+    public function removeComponentGrader($component, $isTeam, $grader_id, $graded_id) {
+        if ($isTeam) {
+            $this->course_db->query("
+            DELETE FROM active_graders
+            WHERE gc_id = ? AND grader_id = ? AND ag_user_id IS NULL AND ag_team_id = ?
+        ", [$component->getId(), $grader_id, $graded_id]);
+        }
+        else {
+            $this->course_db->query("
+            DELETE FROM active_graders
+            WHERE gc_id = ? AND grader_id = ? AND ag_user_id = ? AND ag_team_id IS NULL
```

Comment:
I'm not sure if it would be cleaner or not, but you could combine these into something along the lines of this since the database constraints guarantee that only one of the columns may have a value:
```
WHERE gc_id = ? AND grader_id = ? AND (ag_user_id = ? OR ag_team_id = ?)
```